src/main/java/edu/harvard/iq/dataverse/api/Access.java

```diff
@@ -1829,192 +1807,72 @@ private boolean isAccessAuthorized(DataFile df, String apiToken) {
         } else {
             logger.fine("Session is null.");
         } 
-        
-        User apiTokenUser = null;
-        
-        if ((apiToken != null)&&(apiToken.length()!=64)) {
-            // We'll also try to obtain the user information from the API token, 
-            // if supplied: 
-        
-            try {
-                logger.fine("calling apiTokenUser = findUserOrDie()...");
-                apiTokenUser = findUserOrDie();
-            } catch (WrappedResponse wr) {
-                logger.log(Level.FINE, "Message from findUserOrDie(): {0}", wr.getMessage());
-            }
-            
-            if (apiTokenUser == null) {
-                logger.warning("API token-based auth: Unable to find a user with the API token provided.");
-            }
+        //If we don't have a user, nothing more to do. (Note session could have returned GuestUser)
+        if (user == null && apiTokenUser == null) {
+            logger.warning("Unable to find a user via session or with a token.");
+            return false;
         }
-        
-        // OK, let's revisit the case of non-restricted files, this time in        
-        // an unpublished version:         
+
+        // OK, let's revisit the case of non-restricted files, this time in
+        // an unpublished version:
         // (if (published) was already addressed above)
-        
-        if (!restricted && !embargoed) {
+
+        DataverseRequest dvr = null;
+        if (apiTokenUser != null) {
+            dvr = createDataverseRequest(apiTokenUser);
+        } else {
+            // used in JSF context, user may be Guest
+            dvr = dvRequestService.getDataverseRequest();
+        }
+        if (!published) { // and restricted or embargoed (implied by earlier processing)
```

Comment:
Since published and not restricted/embargoed is handled above, the main split now is whether it is published or not. If it's published, the only case left is with restricted/embargoed. With unpublished, both the restricted/embargoed and not restricted/embargoed both get handled the same way.